src/clj/puppetlabs/services/jruby/puppet_environments.clj

```diff
@@ -36,4 +37,7 @@
       (environment-state [this] state)
       (mark-all-environments-expired! [this]
         (log/info "Marking all registered environments as expired.")
-        (swap! state mark-all-expired!)))))
+        (swap! state mark-all-expired!))
+      (mark-environment-expired! [this env-name]
+        (log/infof "Marking environment '%s' as expired." env-name)
+        (swap! state mark-expired! (keyword env-name))))))
```

Comment:
So if an `env-name` passed in was not previously registered, should this return an error?  Seems like Ruby Puppet typically handles this case by returning an HTTP error with a message about the environment not existing - at least for the endpoints where the environment parameter is actually used.